lib/Backend/LowerMDShared.cpp

```diff
@@ -8790,12 +8804,34 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
                     Assert(src->IsFloat32());
                     pointFive = IR::MemRefOpnd::New((float*)&Js::JavascriptNumber::k_Float32PointFive, TyFloat32, this->m_func, IR::AddrOpndKindDynamicFloatRef);
                 }
-                IR::Instr * addInstr = IR::Instr::New(src->IsFloat64() ? Js::OpCode::ADDSD : Js::OpCode::ADDSS, roundedFloat, roundedFloat, pointFive, this->m_func);
+                this->m_lowerer->InsertCompareBranch(roundedFloat, pointFive, Js::OpCode::BrLt_A, skipAddHalfLabel, instr);
+
+                instr->InsertBefore(addHalfToRoundSrcLabel);
+                IR::RegOpnd* roundedFloatCopy = IR::RegOpnd::New(roundedFloat->GetType(), this->m_func);
+                IR::Instr* roundedFloatBackup = IR::Instr::New(LowererMDArch::GetAssignOp(roundedFloat->GetType()), roundedFloatCopy, roundedFloat, this->m_func);
+                instr->InsertBefore(roundedFloatBackup);
+
+                IR::Instr* roundInstr =
+                    IR::Instr::New(src->IsFloat64() ? Js::OpCode::ROUNDSD : Js::OpCode::ROUNDSS, roundedFloat, roundedFloat, IR::IntConstOpnd::New(0x01, TyInt32, this->m_func), this->m_func);
```

Comment:
I saw your previous comment but when I went back to reply it was gone. Nonetheless, I see your commit to change the operands :)